extensions/parquet/table/src/main/java/io/deephaven/parquet/table/layout/ParquetFileHelper.java

```diff
@@ -0,0 +1,20 @@
+package io.deephaven.parquet.table.layout;
+
+import java.nio.file.FileSystems;
+import java.nio.file.Path;
+import java.nio.file.PathMatcher;
+
+import static io.deephaven.parquet.table.ParquetTableWriter.PARQUET_FILE_EXTENSION;
+
+final class ParquetFileHelper {
+
+    private static final PathMatcher MATCHER;
+
+    static {
+        MATCHER = FileSystems.getDefault().getPathMatcher(String.format("glob:*%s", PARQUET_FILE_EXTENSION));
```

Comment:
This is a whole lot of overkill to check that a filename ends in the right extension. You've turned something that allocates a String into something that allocates the same String and then does regex match, at the cost of additional allocation. At least, that's what `sun.nio.fs.UnixFileSystem#getPathMatcher` looks like.

Reply:
This is an improvement in efficiency wrt how `ParquetFlatPartitionedLayout` was implemented IMO.

Previously, the implementation called `java.nio.file.Files#newDirectoryStream(java.nio.file.Path, java.lang.String)`, which does a regex compilation every call. In this version, we essentially do the regex once.

Reply:
I agree that this is an improvement for `ParquetFlatPartitionedLayout`, but not for other cases where we're using the matcher. You could easily get the best of both worlds if you change the implementation to be:
```
import java.nio.file.Path;

import static io.deephaven.parquet.table.ParquetTableWriter.PARQUET_FILE_EXTENSION;

final class ParquetFileHelper {

    public static boolean fileNameMatches(Path path) {
        return path.getFileName().endsWith(PARQUET_FILE_EXTENSION);
    }
}
```

Reply:
Almost, missing a toString().